common/workunit/workunit.cpp

```diff
@@ -8142,6 +8195,106 @@ bool CLocalWorkUnit::getDebugValueBool(const char * propname, bool defVal) const
+    //of its log lines for a WU into one log file, there is no need for the log file
+    //name to have the 'yyyy_mm_dd' part (10 byte long). The log attribute may look
+    //like '...thormaster.log'.
+    //In the following code, the datePtr is pointed at the end of the 'thormaster.'.
+    //If the strlen(datePtr) < 10, it means that the log file name does not have the
+    //'yyyy_mm_dd' part. If the strlen(datePtr) >= 10, we read the 10 bytes from the
+    //datePtr as the logDate.
+    const char * searchStr = nullptr;
+    if (strieq(processType, "Thor"))
+        searchStr = THORMASTERLOGSEARCHSTR;
+    else
+        searchStr = ECLAGENTLOGSEARCHSTR;
+
+    const char * logFileName = pathTail(logFileNameWithPath);
+    const char * datePtr = strstr(logFileName, searchStr);
+    if (!datePtr)
+    {
+        IWARNLOG("The %s not found in %s.", searchStr, logFileNameWithPath);
+        return logDate;
+    }
+
+    datePtr += strlen(searchStr);
+    if (strlen(datePtr) < 10) //10 bytes for yyyy_mm_dd
+        return logDate;
+
+    logDate.append(10, datePtr);
+    return logDate;
+}
+
+void CLocalWorkUnit::getWUProcessLogInfo(const char *processType, const char *processName, IArrayOf<IConstWUProcessLogInfo> & processLogs) const
+{
+    if (isEmptyString(processType))
+    {
+        IWARNLOG("CLocalWorkUnit::getWUProcessLogInfo(): empty process type.");
+        return;
+    }
+
+    Owned<IPropertyTreeIterator> processInstances = getProcesses(processType, isEmptyString(processName) ? nullptr : processName);
+    ForEach (*processInstances)
+    {
+        StringBuffer logSpec;
+        IPropertyTree &proc = processInstances->query();
+        proc.getProp("@log", logSpec);
+
+        const char *processName = proc.queryName();
+        if (isEmptyString(processName) || logSpec.isEmpty())
+        {
+            IWARNLOG("CLocalWorkUnit::getWUProcessLogInfo(): empty process name or log attribute.");
+            continue;
+        }
+
+        unsigned numberOfThorSlaves = 0, processID = 0;
+        StringBuffer groupName;
+        if (strieq(processType, "Thor"))
+        {
+            //The groupName here may not be used if the WU is imported from a ZAP report because
+            //the getClusterThorGroupName() may not work for the WU imported from a ZAP report.
+            getClusterThorGroupName(groupName, processName);
+
+            numberOfThorSlaves = proc.getPropInt("@max", 0);
+            if (numberOfThorSlaves == 0)
+            { //Legacy WU
+                StringAttr clusterName(queryClusterName());
+                if (clusterName.isEmpty())
+                { //Cluster name may not be set yet
+                    IWARNLOG("Cluster name may not be set for workunit %s", queryWuid());
+                }
+                else
+                {
+                    //This may not work for the WU imported from a ZAP report.
+                    Owned<IConstWUClusterInfo> clusterInfo = getTargetClusterInfo(clusterName);
```

Comment:
should not expect / rely on the cluster being present in the environment, for an imported wuid..